great_expectations/checkpoint/checkpoint.py

```diff
@@ -252,6 +262,21 @@ def _sort_actions(self) -> List[CheckpointAction]:
 
         return priority_actions + secondary_actions
 
+    def is_added(self) -> tuple[bool, list[ResourceNotAddedError]]:
+        errs: list[ResourceNotAddedError] = []
+
+        validations_added: bool = True
+        for validation_definition in self.validation_definitions:
+            validation_added, validation_errs = validation_definition.is_added()
+            errs.extend(validation_errs)
+            validations_added = validation_added and validations_added
```

Comment:
You could factor the top part of this function into it's own one: `are_dependencies_added` and call that here and above. Then you wouldn't need to check for a list with only `CheckpointNotAddedError` above.

Reply:
I was wary of adding too much public surface area since a user working with these objects can just tab complete and see everything on it - perhaps a minor issue but I'd vote to keep it like this for now and clean up later if we desire?

Reply:
Actually if we have `are_dependencies_added`, do we even need `is_added`?

Reply:
My thought here is to has `is_added` return a custom dataclass and then have a property or method on the dataclass that determines whether children are persisted

Reply:
For checkpoints no, but for validation_definitions yes. In terms of autocomplete, I'd just prefix with a `_` since we should only be calling the `_are_dependencies_added` internal to this class.

Reply:
I think this strategy makes error aggregation a little annoying though - if dependencies are not added and neither is the parent, we need to take the output of this function and manually append the `CheckpointNotAddedError`.

```python
deps_added, errors = self._are_dependencies_added()
if not deps_added:
    if not self.id:
       ... # Add additional error
    raise 
```

Reply:
`self.is_added()` already compiles those errors